### PR TITLE
Tasks feature

### PR DESCRIPTION
**Pull request by maj0-0**
Haven't finished the UI part of it. Currently tasks show up in message box like screenshot below
image

Will merge first so wn can start working on mark and unmark, and I'll continue to work on UI

### CODE REVIEW

File: src/main/java/seedu/address/model/group/tasks/TaskInitializer.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+public class TaskInitializer {
+    public static TaskList initializeTasks() throws TaskException {
+        // Create an empty TaskList
+        TaskList taskList = new TaskList();
+
+        List<Task> allTasks = new ArrayList<>();
+
+        // Create tasks for CS2101
+        allTasks.add(new Todo("Upload video of op1", TaskStatus.NOT_DONE, TaskModule.CS2101));
```
Comment (wnchan) [inline src/main/java/seedu/address/model/group/tasks/TaskInitializer.java:15]: Initialiser looks good

**Review by wnchan — COMMENTED** [review]
LGTM!

File: src/main/java/seedu/address/model/group/tasks/TaskInitializer.java (hunk 1 above)
Comment (PearlynnT) [inline src/main/java/seedu/address/model/group/tasks/TaskInitializer.java:15]: Minor nit: 
```suggestion
        allTasks.add(new Todo("Upload video of OP1", TaskStatus.NOT_DONE, TaskModule.CS2101));
```

File: src/main/java/seedu/address/model/group/tasks/TaskInitializer.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        allTasks.add(new Todo("Complete peer review for op1", TaskStatus.NOT_DONE, TaskModule.CS2101));
```
Comment (PearlynnT) [inline src/main/java/seedu/address/model/group/tasks/TaskInitializer.java:16]:
```suggestion
        allTasks.add(new Todo("Complete peer review for OP1", TaskStatus.NOT_DONE, TaskModule.CS2101));
```

File: src/main/java/seedu/address/model/group/tasks/TaskInitializer.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        allTasks.add(new Deadline("Submit slides for op2", TaskStatus.NOT_DONE, TaskModule.CS2101, "29/10/2023 2359"));
```
Comment (PearlynnT) [inline src/main/java/seedu/address/model/group/tasks/TaskInitializer.java:17]:
```suggestion
        allTasks.add(new Deadline("Submit slides for OP2", TaskStatus.NOT_DONE, TaskModule.CS2101, "29/10/2023 2359"));
```

File: src/main/java/seedu/address/model/group/tasks/TaskInitializer.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        allTasks.add(new Deadline("Complete peer review", TaskStatus.NOT_DONE, TaskModule.CS2101, "02/11/2023 2359"));
+        allTasks.add(new Todo("Research on the SCQA framework", TaskStatus.NOT_DONE, TaskModule.CS2101));
+        allTasks.add(new Deadline("plan for op2 ", TaskStatus.NOT_DONE, TaskModule.CS2101, "24/10/2023 2359"));
```
Comment (PearlynnT) [inline src/main/java/seedu/address/model/group/tasks/TaskInitializer.java:20]: Standardise capitalising the first letter of each task
```suggestion
        allTasks.add(new Deadline("Plan for OP2 ", TaskStatus.NOT_DONE, TaskModule.CS2101, "24/10/2023 2359"));
```

File: src/main/java/seedu/address/model/group/tasks/TaskInitializer.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        allTasks.add(new Deadline("submit UG", TaskStatus.NOT_DONE, TaskModule.CS2101, "09/11/2023 2359"));
```
Comment (PearlynnT) [inline src/main/java/seedu/address/model/group/tasks/TaskInitializer.java:21]:
```suggestion
        allTasks.add(new Deadline("Submit UG", TaskStatus.NOT_DONE, TaskModule.CS2101, "09/11/2023 2359"));
```

File: src/main/java/seedu/address/model/group/tasks/TaskInitializer.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+        // Create tasks for CS2103T
+        allTasks.add(new Todo("complete mid semester review form", TaskStatus.NOT_DONE, TaskModule.CS2103T));
```
Comment (PearlynnT) [inline src/main/java/seedu/address/model/group/tasks/TaskInitializer.java:24]:
```suggestion
        allTasks.add(new Todo("Complete mid semester review form", TaskStatus.NOT_DONE, TaskModule.CS2103T));
```

File: src/main/java/seedu/address/model/group/tasks/TaskInitializer.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        allTasks.add(new Deadline("add demo screenshots to project notes", TaskStatus.NOT_DONE, TaskModule.CS2103T, "20/10/2023 2359"));
```
Comment (PearlynnT) [inline src/main/java/seedu/address/model/group/tasks/TaskInitializer.java:25]:
```suggestion
        allTasks.add(new Deadline("Add demo screenshots to project notes", TaskStatus.NOT_DONE, TaskModule.CS2103T, "20/10/2023 2359"));
```

File: src/main/java/seedu/address/model/group/tasks/TaskInitializer.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        allTasks.add(new Deadline("release v1.3.trial jar file", TaskStatus.NOT_DONE, TaskModule.CS2103T, "27/10/2023 2359"));
```
Comment (PearlynnT) [inline src/main/java/seedu/address/model/group/tasks/TaskInitializer.java:26]:
```suggestion
        allTasks.add(new Deadline("Release v1.3.trial jar file", TaskStatus.NOT_DONE, TaskModule.CS2103T, "27/10/2023 2359"));
```

File: src/main/java/seedu/address/model/group/tasks/TaskInitializer.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+        allTasks.add(new Deadline("wrap up mileston 1.3", TaskStatus.NOT_DONE, TaskModule.CS2103T, "03/11/2023 2359"));
```
Comment (PearlynnT) [inline src/main/java/seedu/address/model/group/tasks/TaskInitializer.java:27]: Typo here
```suggestion
        allTasks.add(new Deadline("Wrap up milestone 1.3", TaskStatus.NOT_DONE, TaskModule.CS2103T, "03/11/2023 2359"));
```

File: src/main/java/seedu/address/model/group/tasks/TaskInitializer.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        allTasks.add(new Deadline("finalise tp", TaskStatus.NOT_DONE, TaskModule.CS2103T, "17/11/2023 2359"));
```
Comment (PearlynnT) [inline src/main/java/seedu/address/model/group/tasks/TaskInitializer.java:28]:
```suggestion
        allTasks.add(new Deadline("Finalise tp", TaskStatus.NOT_DONE, TaskModule.CS2103T, "17/11/2023 2359"));
```

File: src/main/java/seedu/address/model/group/tasks/TaskInitializer.java [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+        allTasks.add(new Todo("update dg for each feature", TaskStatus.NOT_DONE, TaskModule.CS2103T));
```
Comment (PearlynnT) [inline src/main/java/seedu/address/model/group/tasks/TaskInitializer.java:29]:
```suggestion
        allTasks.add(new Todo("Update DG for each feature", TaskStatus.NOT_DONE, TaskModule.CS2103T));
```

**Review by PearlynnT — COMMENTED** [review]
Minor wording nits

**Pull request merged**